File: Level 0/CSharp Basics May 2014/ExamProblems/Exam12Apr14Morning_01/Triangle.cs
using System;

class Triangle
{
    static void Main()
    {
        int x1 = int.Parse(Console.ReadLine());
        int y1 = int.Parse(Console.ReadLine());
        int x2 = int.Parse(Console.ReadLine());
        int y2 = int.Parse(Console.ReadLine());
        int x3 = int.Parse(Console.ReadLine());
        int y3 = int.Parse(Console.ReadLine());

        double a = Math.Sqrt((x2 - x1) * (x2 - x1) + (y2 - y1) * (y2 - y1));
        double b = Math.Sqrt((x3 - x2) * (x3 - x2) + (y3 - y2) * (y3 - y2));
        double c = Math.Sqrt((x1 - x3) * (x1 - x3) + (y1 - y3) * (y1 - y3));

        if (a + b > c && b + c > a && a + c > b)
        {
            Console.WriteLine("Yes");
            double p = (a + b + c) / 2;
            double area = Math.Sqrt(p * (p - a) * (p - b) * (p - c));
            Console.WriteLine("{0:F2}", area);
        }
        else
        {
            Console.WriteLine("No");
            Console.WriteLine("{0:F2}", a);
        }
    }
}

File: Level 2/01. OOP (January 2015)/09. OOPinJava/src/Geometry/Shapes/PlaneShapes/Triangle.java
package Geometry.Shapes.PlaneShapes;

public class Triangle extends PlaneShape {

    public Triangle(double[] firstVertice, double[] secondVertice, double[] thirdVertice) throws Exception{
        this.setVertices(firstVertice);
        this.setVertices(secondVertice);
        this.setVertices(thirdVertice);
    }

    @Override
    public double getArea() {
        double aX = this.getVertices().get(0)[0];
        double aY = this.getVertices().get(0)[1];
        double bX = this.getVertices().get(1)[0];
        double bY = this.getVertices().get(1)[1];
        double cX = this.getVertices().get(2)[0];
        double cY = this.getVertices().get(2)[1];
        double area = Math.abs((aX * (bY - cY) + bX * (cY - aY) + cX * (aY - bY)) / 2);

        return area;
    }

    @Override
    public double getPerimeter() {
        double sideA = distanceBetweenTwoPoints(this.vertices.get(0), this.vertices.get(1));
        double sideB = distanceBetweenTwoPoints(this.vertices.get(1), this.vertices.get(2));
        double sideC = distanceBetweenTwoPoints(this.vertices.get(2), this.vertices.get(0));
        double perimeter = sideA + sideB + sideC;

        return perimeter;
    }

    @Override
    public String toString() {

        String result = "Triangle:\n";

        for (int i = 0; i < this.getVertices().size(); i++) {
            result += String.format("Point%d: x=%.2f y=%.2f\n", i + 1, this.getVertices().get(i)[0], this.getVertices().get(i)[1]);
        }

        result += String.format("Area: %.2f\n", this.getArea());
        result += String.format("Perimeter: %.2f\n", this.getPerimeter());

        return result;
    }

    private double distanceBetweenTwoPoints(double[] firstVertice, double[] secondVertice){

        double distance = Math.sqrt(Math.pow(firstVertice[0] - secondVertice[0], 2) + Math.pow(firstVertice[1] - secondVertice[1], 2));

        return distance;
    }
}